Zilu fix

MartinSpiessl opened the pull request:
This reapplies zhe zilu benchmarks, which were reverted with https://github.com/sosy-lab/sv-benchmarks/pull/1172 because at the time the task were originally merged, the `diff_ci.sh` contained a bug and the missing declarations of `__VERIFIER_nondet` functions in the tasks went unnotticed.

This PR reapplies the tasks and also adds the missing declarations, which will (hopefully) get rid of the compile errors.

I will mark this PR as draft because @hernanponcedeleon found some overflows which first need to be fixed now:
https://github.com/sosy-lab/sv-benchmarks/issues/1159

MartinSpiessl commented:
I forgot to remove the draft status yesterday (or I remember I couldn't find the button before the CI finished?)

Ready for review now

dbeyer commented:
Which reviewer did you ask to review it?

MartinSpiessl commented:
I marked it as `help wanted` because there isn't yet a reviewer

MartinSpiessl commented:
Hmm would make sense to ask @hernanponcedeleon

hernanponcedeleon commented:
I'll try to do it during the week.
Just something I spot after a quick look: benchmarks for the reachability category don't have a `no-overflow` property (which should be true). Can you please add them?

MartinSpiessl commented:
Sure, just added it, thanks for the hint! I originally just used some `sed` magic to change the verdicts, now I updated my task generation script to output the proper verdict.

hernanponcedeleon commented:
I think all the following overflows are possible:

`benchmark03`: `x` and `y` can overflow after `MAX_INT` iterations. `i` and `j` before.
`benchmark08`: `sum` can overflow, e.g. if `n = MAX_INT`
`benchmark13`: `i` can overflow, e.g. if `k = MAX_INT`
`benchmark16`: `i` can overflow after `MAX_INT-1` iterations.
`benchmark22`: `y` can overflow after `MAX_INT` iterations. `x` even before.
`benchmark24`: `i` can overflow for any `n > MAX_INT/2` iterations.
`benchmark36`: `x` and `y` can overflow after `MAX_INT` iterations.
`benchmark38`: `y` can overflow after `MAX_INT` iterations. `x` after `MAX_INT/4+1` iterations
`benchmark41`:: `x` and `y` can overflow after `MAX_INT` iterations. `z` after `MAX_INT/2+1` iterations

hernanponcedeleon reviewed (changes requested):
See above

MartinSpiessl commented:
Thanks, I just updated the verdicts according to your listing. I checked the first few and trust your reporting to be correct. If not then the verifiers will hopefully find this anyway.

hernanponcedeleon reviewed (approved):
My comments were based on manual inspection of the benchmarks.
I tried to run `cbmc` to check for overflows, but since they all require ~`MAX_INT` iterations, this was not feasible.
I tried the benchmarks were we claim `no-overflow = true` for reachability with Dartagnan and `-O3` compiler options (this gave me hints in the past about undefined behaviours) and I don't get any wrong results. This makes me believe at least those expected results are ok.
